### PR TITLE
fix(drawer): solve the focus caused element jump bug #3703

### PR DESCRIPTION
**Pull request by ioslh**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?
This PR is mainly  about two issues.

## Fix a long existing bug caused by `focus`

Solve the `focus` caused element jump first time drawer opening, there is an issue concerning this bug #3703 .

- At the first time drawer opening, we set `visible` to `true`, which will call `render` twice(maybe another issue needs to be solved). After the first callling drawer is not truly shown in container([getChildToRender called by false](https://github.com/vueComponent/ant-design-vue/blob/next/components/vc-drawer/src/Drawer.js#L595)), but [domFocus](https://github.com/vueComponent/ant-design-vue/blob/next/components/vc-drawer/src/Drawer.js#L85) is triggered and called.
- What happened if you call `focus` on an invisible element?  Please refer to this [codepen](https://codepen.io/ioslh/pen/gORrJrJ). `focus` [internally called `scrollIntoViewIfNeeded`](https://chromium.googlesource.com/chromium/src.git/+/refs/heads/main/third_party/blink/renderer/core/dom/element.cc#4231) , that's why the element jump weirdly, unless you pass `{ preventScroll: true}`  to `focus` as parameter.

And [this example](https://2x.antdv.com/components/drawer-cn#components-drawer-demo-render-in-current) don't have to implemented in that tricky way.

## Add `examples` folder to `.gitignore`
Maybe every maintainer write different example code temporary, don't have to commit that every time, keep in local machine would be better.

### CODE REVIEW

**Comment by tangjinzhou** [comment]
Keep the examples document for now. We will open source all documents soon

Can it be solved by `settimeout`

File: components/vc-drawer/src/Drawer.js [hunk 1]
```diff
@@ -142,7 +142,7 @@ const Drawer = defineComponent({
   methods: {
     domFocus() {
       if (this.dom) {
-        this.dom.focus();
+        this.dom.focus({ preventScroll: true });
```
Comment (ajuner) [inline components/vc-drawer/src/Drawer.js:145]: Is it better to change to `setTimeout` where domFocus is triggered?
https://github.com/vueComponent/ant-design-vue/blob/dd859500c2d8c58f1f12dd46ba3330cceee3bd30/components/vc-drawer/src/Drawer.js#L84

Comment (ioslh) [reply components/vc-drawer/src/Drawer.js:145]: I tried, it actually work. Given we found the root cause is calling `focus` on invisible element without preventScroll, we can fix it by
- pass `preventScroll` to `focus`
- or guarantee element appeared in viewport before `focus` called, so yes `setTimeout` would do the job, but I'm not sure will it cause any other side effects.

And I personally always consider `setTimeout` to be the last resort to resolve issues if you have any other options.

Comment (tangjinzhou) [reply components/vc-drawer/src/Drawer.js:145]: you are right.  but preventScroll will cause breaking change( we need scrollIntoView).
For 2.x, we can use setTimeout resolve it .  we will refactor render logic in v3.x.

**Pull request closed without merging**